### ISSUE TITLE
Windows: error C2036: 'mapnik::glyph_info *': unknown size

### ISSUE THREAD

**Issue by BergWerkGIS** [mapnik/mapnik#2971]
Both, with 2014CTP4 and 2015RC1

[MSDN: Compiler Error C2036](https://msdn.microsoft.com/en-us/library/1kay26wa(v=vs.140).aspx)

```
 "c:\mb\windows-builds-64\packages\mapnik-master\mapnik-gyp\build\mapnik.sln" (default target) (1) ->
       "c:\mb\windows-builds-64\packages\mapnik-master\mapnik-gyp\build\csv.vcxproj.metaproj" (default target) (2) ->
       "c:\mb\windows-builds-64\packages\mapnik-master\mapnik-gyp\build\mapnik.vcxproj" (default target) (16) ->
       (ClCompile target) ->
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vector(1650): error C2036: 'mapnik::glyph_info *': unknown
        size (compiling source file ..\..\src\agg\process_shield_symbolizer.cpp) [c:\mb\windows-builds-64\packages\mapnik-master\m
       apnik-gyp\build\mapnik.vcxproj]
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vector(1650): error C2036: 'mapnik::glyph_info *': unknown
        size (compiling source file ..\..\src\agg\process_text_symbolizer.cpp) [c:\mb\windows-builds-64\packages\mapnik-master\map
       nik-gyp\build\mapnik.vcxproj]
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\vector(1650): error C2036: 'mapnik::glyph_info *': unknown
        size (compiling source file ..\..\src\agg\process_group_symbolizer.cpp) [c:\mb\windows-builds-64\packages\mapnik-master\ma
       pnik-gyp\build\mapnik.vcxproj]
```

**Comment by springmeyer** [comment]
looks like a missing include such that the full definition of that class is not available. thanks for reporting.

**Comment by springmeyer** [comment]
@BergWerkGIS - pushed a new branch (https://github.com/mapnik/mapnik/compare/issue-2971) that I hope will fix this. Can you give that a shot?

**Comment by BergWerkGIS** [comment]
@springmeyer 

```
       "C:\mb\windows-builds-32\packages\mapnik-issue-2971\mapnik-gyp\build\mapnik.sln" (default target) (1) ->
       "C:\mb\windows-builds-32\packages\mapnik-issue-2971\mapnik-gyp\build\csv.vcxproj.metaproj" (default target) (2) ->
       "C:\mb\windows-builds-32\packages\mapnik-issue-2971\mapnik-gyp\build\mapnik.vcxproj" (default target) (16) ->
       (ClCompile target) ->
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xmemory0(659): error C2280: 'mapnik::text_itemizer::run<UB
       iDiDirection>::run(const mapnik::text_itemizer::run<UBiDiDirection> &)': attempting to reference a deleted function (compil
       ing source file ..\..\src\agg\process_shield_symbolizer.cpp) [C:\mb\windows-builds-32\packages\mapnik-issue-2971\mapnik-gyp
       \build\mapnik.vcxproj]
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xmemory0(659): error C2280: 'mapnik::text_itemizer::run<UB
       iDiDirection>::run(const mapnik::text_itemizer::run<UBiDiDirection> &)': attempting to reference a deleted function (compil
       ing source file ..\..\src\agg\process_text_symbolizer.cpp) [C:\mb\windows-builds-32\packages\mapnik-issue-2971\mapnik-gyp\b
       uild\mapnik.vcxproj]
         C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\include\xmemory0(659): error C2280: 'mapnik::text_itemizer::run<UB
       iDiDirection>::run(const mapnik::text_itemizer::run<UBiDiDirection> &)': attempting to reference a deleted function (compil
       ing source file ..\..\src\agg\process_group_symbolizer.cpp) [C:\mb\windows-builds-32\packages\mapnik-issue-2971\mapnik-gyp\
       build\mapnik.vcxproj]

```

**Comment by BergWerkGIS** [comment]
@springmeyer possible hint: https://github.com/mapnik/mapnik/issues/2487

**Comment by BergWerkGIS** [comment]
\o/ compiled on VS2014 CTP4 64bit.
Haven't tried any other variations yet.

**Comment by BergWerkGIS** [comment]
Yes, this compiles also with VS2015 release version, `master` does not.

**Comment by springmeyer** [comment]
merged #3002 which fixes this issue. The downside is that this merge will make it hard to revert https://github.com/mapnik/mapnik/commit/3ed7e347cb7f43c7685c282fae6a9069d978dc14 (the commit that I think triggered this). However the changes in this branch are fine overall and if I choose to revert https://github.com/mapnik/mapnik/commit/3ed7e347cb7f43c7685c282fae6a9069d978dc14 I'll document that decision at #2965
